spi/src/main/java/com/amazon/opendistroforelasticsearch/jobscheduler/spi/utils/LockService.java

```diff
@@ -306,6 +306,32 @@ public void deleteLock(final String lockId, ActionListener<Boolean> listener) {
                 }));
     }
 
+    /**
+     * Attempt to renew a lock.
+     * It is used to give an extended valid period to a lock. The start time of the lock will be updated to
+     * the current time when the method get called, and the duration of the lock remains.
+     * It works as long as the lock is not acquired by others, and no matter if the lock is expired of not.
+     *
+     * @param lock a {@code LockModel} to be renewed.
+     * @param listener a {@code ActionListener} that has onResponse and onFailure that is used to return whether
+     *                 or not the renewal was successful
+     */
+    public void renewLock(final LockModel lock, ActionListener<Boolean> listener) {
+        if (lock == null) {
+            logger.debug("Lock is null. Nothing to renew.");
+            listener.onResponse(false);
+        } else {
+            logger.debug("Renewing lock: {}. " +
+                            "The lock was acquired or renewed on: {}, and it will be valid for another {} sec from now.",
+                    lock, lock.getLockTime(), lock.getLockDurationSeconds());
+            final LockModel lockToRenew = new LockModel(lock, getNow(), lock.getLockDurationSeconds(), false);
+            updateLock(lockToRenew, ActionListener.wrap(
+                    renewedLock -> listener.onResponse(renewedLock != null),
```

Comment:
I think it might be more useful to change this to return the lock itself rather than a boolean as it would be more useful to have the new updated lock in the plugin with the new seqNo/primaryTerm.

Reply:
Ok, I will change to return the lock. I have realized it could be more useful as well during writing the test.

Reply:
I have changed to return the lock object instead of a boolean value after calling the method, and also simplified the test procedure due to the change.